tech-specs/src/explore/design/dkg.md

```diff
@@ -1 +1,178 @@
-# Distributed key generation gossip
+# Distributed key generation
+
+Here we give an overview of how the DKG protocol works with the ledger and 
+the roles of each node in the network per this protocol. This portion of the 
+book focuses solely on the production of a new public encryption key via the
+DKG scheme. It does not cover encryption, decryption, or validation of transactions.
+
+Some parts of the DKG protocol may be done through the gossip layer or through
+off-chain communication. However, the entire DKG cannot exist solely in the 
+gossip layer as consensus on the final shared secrets must be achieved. This 
+will require posting information on chain.
+
+## Starting the Protocol
+
+The protocol to generate a new key will be scheduled to begin at specific 
+block heights, in pratices this will be at the beginning of every eposh.
+This way all the validators know when to begin the announcement phase. In 
+addition to starting at the pre-determined blocks heights, if the protocol 
+fails, all validators know to immediately start a new round of the protocol.
+
+It is important to note that the key that is being generated is to be used
+encrypting transactions in the epoch following the current one. However,
+this relies on the assumption that after the validators sets change between
+epochs that there is still validators whose combined weight exceeds 
+\\( \frac{2}{3} \\) of \\W\\. Otherwise, it will be impossible to decrypt
+transactions. If this does not happen a new key will need to be generated.
+
+__TBD__: Do we accept unencrypted transactions while the new key is generated
+or do we stop accepting transactions until the new key is available? Or 
+something else (that is hopefully smarter)?
+
+##  Distributing weight shares
+
+In order for the DKG protocol to work, it requires the contribution of 
+ \\(\frac{2}{3} \\) of the total staked weight among validators. At the
+beginning of the DKG protocol, Tendermint can be queried each validator
+to get the set of validators, their associated address, and voting power
+(or weight).
+
+The total weight will be scaled down to a fixed parameter 
+\\( W\\) (and fractional weights rounded appropriately to achieve integral
+values). After this procedure, each validator has a corresponding number of
+weight shares. 
+
+The DKG protocol requires a canonical ordering of validators. The ordering
+will be chosen by ordering validators in terms of decreasing voting power.
+If there is a tie, it is broken by considering the lexicographic ordering
+on their associated addresses. 
+
+This allows each validator to independently compute the partition of weight
```

Comment:
not sure exactly what this sentence means, do you mean that each validator computes how many shares they get? how many shares all of the other validators get?

Reply:
Everyone computes the shares that everyone has. And by everyone, I mean validators. Tried to clarify.